src/sentry/api/endpoints/project_symbol_sources.py

```diff
@@ -0,0 +1,353 @@
+from uuid import uuid4
+
+from drf_spectacular.utils import extend_schema
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NO_CONTENT,
+    RESPONSE_NOT_FOUND,
+)
+from sentry.apidocs.examples.project_examples import ProjectExamples
+from sentry.apidocs.parameters import GlobalParams, ProjectParams
+from sentry.lang.native.sources import (
+    REDACTED_SOURCE_SCHEMA,
+    REDACTED_SOURCES_SCHEMA,
+    VALID_CASINGS,
+    VALID_FILE_TYPES,
+    VALID_LAYOUTS,
+    InvalidSourcesError,
+    backfill_source,
+    parse_sources,
+    redact_source_secrets,
+    validate_sources,
+)
+from sentry.models.project import Project
+from sentry.utils import json
+
+
+class LayoutSerializer(serializers.Serializer):
+    type = serializers.ChoiceField(
+        choices=VALID_LAYOUTS, help_text="The source's layout type.", required=True
+    )
+    casing = serializers.ChoiceField(
+        choices=VALID_CASINGS, help_text="The source's casing rules.", required=True
+    )
+
+
+class SourceSerializer(serializers.Serializer):
+    type = serializers.ChoiceField(
+        choices=["appStoreConnect", "http", "gcs", "s3"],
+        required=True,
+        help_text="The type of the source.",
+    )
+    id = serializers.CharField(
+        required=False,
+        help_text="The internal ID of the source. Must be distinct from all other source IDs and cannot start with 'sentry:'. If this is not provided, a new UUID will be generated.",
+    )
+    name = serializers.CharField(
+        required=True,
+        help_text="The human-readable name of the source.",
+    )
+    filetypes = serializers.ListField(
```

Comment:
I looked on the frontend for this but it doesn't look like we allow users to set this field themselves. I found some [docs](https://docs.sentry.io/platforms/apple/guides/ios/data-management/debug-files/symbol-servers/#directory-layouts) that have a chart with file types, might be useful?

Reply:
You're right, there seems to be no way to set this. I think I'll just remove it.